templates/shared/download/pdf_layout.html.twig

```diff
@@ -113,7 +113,7 @@
     <table class="layout">
         <tr>
             <td>
-                <img width="160" src="{{ logoPath }}">
+                <img width="160" src="{{ generateWithGotenberg ? gotenberg_asset(logoPath) : logoPath }}">
```

Comment:
inject the config key `$config['pdf_generator']['generate_with_gotenberg']` to wrap this logic in an if maybe ?